src/Sentry.GraphQl/SentryGraphQlHttpFailedRequestHandler.cs

```diff
@@ -0,0 +1,93 @@
+namespace Sentry.GraphQl;
+
+internal class SentryGraphQlHttpFailedRequestHandler : SentryFailedRequestHandler
+{
+    private readonly IHub _hub;
+    private readonly SentryOptions _options;
+    private readonly GraphQlContentExtractor _extractor;
+    internal const string MechanismType = "SentryGraphQLHttpFailedRequestHandler";
+    private readonly SentryHttpFailedRequestHandler _httpFailedRequestHandler;
+
+    internal SentryGraphQlHttpFailedRequestHandler(IHub hub, SentryOptions options, GraphQlContentExtractor? extractor = null)
```

Comment:
```suggestion
    public SentryGraphQlHttpFailedRequestHandler(IHub hub, SentryOptions options, GraphQlContentExtractor? extractor = null)
```
That's the constructor to be used, right?

Reply:
It is, but the only place it would ever get used is in the internal constructor for [`SentryGraphQlHttpMessageHandler`](https://github.com/getsentry/sentry-dotnet/blob/ef28a1a82856f767ae667de25d02b60eac91210b/src/Sentry.GraphQl/SentryGraphQlHttpMessageHandler.cs#L34), which is basically just there for testings. 

We don't presently allow SDK users to pass in their own failed request handler to our message handlers.

One reason for keeping the failed request handlers separate from the message handlers themselves (since they might look, at first glance, tightly coupled) is that they are a bit more composable this way... and you can see I've actually used that composability [here](https://github.com/getsentry/sentry-dotnet/blob/ef28a1a82856f767ae667de25d02b60eac91210b/src/Sentry.GraphQl/SentryGraphQlHttpFailedRequestHandler.cs#L36). If the `GraphQLFailedRequestHandler` doesn't find any graphql specific errors in the response content, it passes the response for processing by the `SentryHttpFailedRequestHandler` to check if we received a status code that would indicate some other kind of failure (e.g. a 500 internal server error).